### PR TITLE
FOGL-528: User interface for backup/restore

### PR DESCRIPTION
**Pull request by monika-Dianomic**
- [x] tested with https + auth mandatory
- [x] tested with no backup exists and other failures simulation

**NOTE:** Restore backup has an issue on API side **(FOGL-834)**

### CODE REVIEW

File: src/app/backup-restore/backup-restore.component.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    );
+  }
+
+  public requestBackup() {
+    this.ngProgress.start();
+    this.backupRestoreService.requestBackup().
+    subscribe(
+      data => {
+        this.ngProgress.done();
+        this.alertService.success("Backup created successfully");
```
Comment (praveen-garg) [inline src/app/backup-restore/backup-restore.component.ts]: I think  we should display whatever message is returned from api

File: src/app/backup-restore/backup-restore.component.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    );
+  }
+
+  public restoreBackup(id) {
+    this.ngProgress.start();
+    this.backupRestoreService.restoreBackup(id).
+    subscribe(
+      data => {
+        this.ngProgress.done();
+        this.alertService.success("Restored successfully");
```
Comment (praveen-garg) [inline src/app/backup-restore/backup-restore.component.ts]: I think  we should display whatever message is returned from api

File: src/app/backup-restore/backup-restore.component.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    );
+  }
+
+  public deleteBackup(id) {
+    this.ngProgress.start();
+    this.backupRestoreService.deleteBackup(id).
+    subscribe(
+      data => {
+        this.ngProgress.done();
+        this.alertService.success("Backup deleted successfully");
```
Comment (praveen-garg) [inline src/app/backup-restore/backup-restore.component.ts]: I think  we should display whatever message is returned from api

File: src/app/backup-restore/backup-restore.component.ts (hunk 1 above)
Comment (monika-Dianomic) [reply src/app/backup-restore/backup-restore.component.ts]: It is returning `{"status": "running"}`

File: src/app/backup-restore/backup-restore.component.ts (hunk 2 above)
Comment (monika-Dianomic) [reply src/app/backup-restore/backup-restore.component.ts]: It is returning `{"status": "running"}`

File: src/app/backup-restore/backup-restore.component.ts (hunk 1 above)
Comment (praveen-garg) [reply src/app/backup-restore/backup-restore.component.ts]: Yes please, if needs to be fixed, it should be on API side

File: src/app/backup-restore/backup-restore.component.ts (hunk 2 above)
Comment (praveen-garg) [reply src/app/backup-restore/backup-restore.component.ts]: Yes please, if needs to be fixed, it should be on API side

File: src/app/services/backup-restore.service.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs/Rx';
```
Comment (mshariq-nerd) [inline src/app/services/backup-restore.service.ts]: `import { Observable } from 'rxjs/Observable' instead of importing full library.

**Pull request merged**